src/test/java/com/cloudant/tests/ViewsTest.java

```diff
@@ -107,13 +107,227 @@ public void byStartAndEndKey() {
+                .endKey(false);
+
+        List<Object> result = query.query(Object.class);
+
+        assertThat(result.size(), is(3));
+
+        Page page = null;
+        try {
+            page = query.queryPage(30, null, Object.class);
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+
+        assertThat(page.getResultList().size(), is(3));
+    }
+
+    /**
+     * Assert that passing a string key with double quotes
+     * or spaces in query and queryView will produce
+     * a result list without exception.
+     */
+    @Test
+    public void byStartAndEndStringKey() {
+        init();
+
+        //Query for testing string with spaces
+        View query = db.view("example/spaces_created")
+                .startKey(" spaces ", 1)
+                .endKey(" spaces 2", 2000);
+
+        List<Object> result = query.query(Object.class);
+
+        assertThat(result.size(), is(3));
+
+        Page page = null;
+        try {
+            page = query.queryPage(30, null, Object.class);
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+
+        assertThat(page.getResultList().size(), is(3));
+
+        //Query for testing string with quotes
+        query = db.view("example/quotes_created")
+                .startKey("\"quotes\" ", 1)
+                .endKey("\"quotes\" 3", 2000);
+
+        result = query.query(Object.class);
+
+        assertThat(result.size(), is(3));
+
+        page = null;
+        try {
+            page = query.queryPage(30, null, Object.class);
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+
+        assertThat(page.getResultList().size(), is(3));
+    }
+
+    /**
+     * Assert that passing a single integer value key expression
+     * in query and queryView will produce a result list of all docs.
+     *
+     * Run the same assert above but with a key that is an object array
+     * containing an integer.
+     */
+    @Test
+    public void byStartAndEndIntKeyArray() {
+        init();
+
+        View query = db.view("example/created")
+                .startKey(1)
+                .endKey(2000);
+
+        List<Object> result = query.query(Object.class);
+
+        assertThat(result.size(), is(3));
+
+        Page page = null;
+        try {
+            page = query.queryPage(30, null, Object.class);
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+
+        assertThat(page.getResultList().size(), is(3));
+
+        query = db.view("example/created")
+                .startKey(new Object[]{1})
+                .endKey(new Object[]{2000});
+
+
+        result = query.query(Object.class);
+
+        assertThat(result.size(), is(3));
+    }
+
+    /**
+     * Assert that passing a key with two values in query
+     * and queryView will produce a result list of all docs.
+     *
+     * Run the same assert above but with a key that is an
+     * object array containing only integers.
```

Comment:
Think this should be changed to only **numbers** since the code says `.startKey(1, 12.00)` which is a mix of int and float (but all numbers from a JSON perspective).